pkg/plugins/resources/memory/store.go

```diff
@@ -151,6 +151,21 @@ func (c *memoryStore) Delete(_ context.Context, r model.Resource, fs ...store.De
 	}
 	return nil
 }
+func (c *memoryStore) DeleteMany(_ context.Context, fs ...store.DeleteManyOptionsFunc) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	opts := store.NewDeleteManyOptions(fs...)
+
+	retained := memoryStoreRecords{}
+	for _, record := range c.records {
+		if opts.Mesh != "" && opts.Mesh != record.Mesh {
```

Comment:
Support for `opts.Mesh == ""` is too dangerous (I don't think somebody will ever need this feature)